packages/next-server/server/next-server.js

```diff
@@ -165,6 +165,10 @@ export default class Server {
   }
 
   async render (req, res, pathname, query, parsedUrl) {
+    if (isResSent(res)) {
```

Comment:
In what case does this happen?

Reply:
Programmer error that calls `app.render()` after `res.end()` or `res.flushHeaders()`. Since the check was already there I thought it was a good idea to move it at the top.

Reply:
I don't think we should make this change as it obfuscates bugs in the user's code, if someone calls `app.render` after the response has ended they should prevent that from happening.

Reply:
Agreed, but the point is that you are already obfuscating it in current code. In fact no error is thrown if something like that is done and instead CPU cycles are wasted rending stuff which shouldn't be rendered.

Reply:
Then let's throw an error when the res has been sent 👍

Reply:
Feel free to do that. I'm not passionate enough to open a new PR and add a test atm. Maybe I will do that if I can spare some time one of the coming days.